cmd/gapit/common.go

```diff
@@ -52,11 +52,21 @@ func (f CommandFilterFlags) commandFilter(ctx context.Context, client service.Se
 		if err != nil {
 			return nil, log.Err(ctx, err, "Failed to load the contexts")
 		}
-		contexts := boxedContexts.(*service.Contexts)
-		if n := len(contexts.List); f.Context >= n {
-			return nil, log.Errf(ctx, err, "Context %d is out of range [0..%d]", f.Context, n-1)
+		contexts := boxedContexts.(*service.Contexts).List
+
+		for i, c := range contexts {
+			boxedContext, err := client.Get(ctx, p.Context(c.ID.ID()).Path(), nil)
+			if err != nil {
+				return nil, log.Errf(ctx, err, "Failed to load context at index %d", i)
+			}
+			context := boxedContext.(*service.Context)
+			if f.Context == int(context.Handle) {
```

Comment:
Would it be possible to not introduce a `Handle`, and just compare f.Context with context.ID with some proper casting / stringification if need be?

Reply:
Having a deeper look, this is where a plain gles context identifier gets translated to a ContexID:

https://github.com/google/gapid/blob/46511520bc56e6053b1ecdb3f46476361f476f6a/gapis/api/gles/context.go#L38

Maybe this logic should be abstracted to an exported function, such that in `commands.go` you can re-compute the ContextID from the f.Context integer using the same logic, to be able to compare them, without introducing a handle.

Reply:
We should not do that last bit. `commands.go` should not have any knowledge about GLES or Vulkan to speak of. This one of the main advantages of the RPC infrastructure.

If we want to do something similar, we should export the name of the context through a `path`, then we can get them in a single place.

Reply:
Actually reading this a bit more, scratch that last comment. We already have a name in the service.Context.
I think you may run into problems with context.ID. What happens in GLES if you create/destroy/create a context. Are you guaranteed to get a new unique handle? I expect the spec makes no such requirement. If that is the case we should strongly consider just ordering the contexts in the list on the server-side. Alternatively (or additionally) we could specify the context by string if we want that.

--context_name "Context 2" which would match by name.

Reply:
I don't know if ordering will work since you can have a 1 context without a 0. I have traces like this.

Reply:
Alright, then I would prefer to leave as-is if we have to, but add the --context_name feature.

Baking GL information into gapit is not something we really want to do. It also means we cannot change things more flexibly in the future.